src/main/java/org/ohdsi/circe/check/checkers/NoExitCriteriaCheck.java

```diff
@@ -37,9 +37,12 @@ protected WarningSeverity defineSeverity() {
 
     @Override
     protected void check(CohortExpression expression, WarningReporter reporter) {
-
         match(expression)
-                .when(e -> Objects.equals("All", e.primaryCriteria.primaryLimit.type) && Objects.isNull(e.endStrategy))
+                .when(e -> Objects.equals("All", e.primaryCriteria.primaryLimit.type)
+                        && Objects.isNull(e.endStrategy)
+                        && Objects.equals("All", e.expressionLimit.type)
+                        && (Objects.isNull(e.additionalCriteria)
+                            || Objects.equals("All", e.qualifiedLimit.type)))
```

Comment:
we use 'equalsIgnoreCase' in many places, i apologize that we didn't put these coded values in some kind of constants, but that's the world we're currently living in. I plan on remediating some of this in Circe 2.0.

Reply:
fixed